forge-gui/res/cardsfolder/upcoming/guff_rewrites_history.txt

```diff
@@ -0,0 +1,13 @@
+Name:Guff Rewrites History
+ManaCost:2 R
+Types:Instant 
+A:SP$ ChangeZone | ValidTgts$ Permanent.nonLand+nonEnchantment | RememberChanged$ True | TgtPrompt$ For each player, choose target nonenchantment, nonland permanent that player controls. | Origin$ Battlefield | Destination$ Library | Shuffle$ True | TargetMin$ 0 | TargetMax$ OneEach | TargetsWithDifferentControllers$ True | SubAbility$ DBRepeat | SpellDescription$ For each player, choose target nonenchantment, nonland permanent that player controls. Those permanents' owners shuffle them into their libraries. Each player who controlled one of those permanents exiles cards from the top of their library until they exile a nonland card, then puts the rest on the bottom of their library in a random order. Each player may cast the nonland card they exiled without paying its mana cost.
+SVar:DBRepeat:DB$ RepeatEach | RepeatPlayers$ RememberedController | RepeatSubAbility$ DBDigUntil | SubAbility$ DBRepeatCard
+SVar:DBDigUntil:DB$ DigUntil | Defined$ RememberedController | Valid$ Permanent.nonLand | FoundDestination$ Exile | RevealedDestination$ Exile | ImprintFound$ True | RememberRevealed$ True | SubAbility$ DBChange
+SVar:DBChange:DB$ ChangeZoneAll | ChangeType$ Card.IsRemembered | Origin$ Exile | Destination$ Library | RandomOrder$ True | LibraryPosition$ -1 | SubAbility$ DBCleanup
```

Comment:
You don't want to move the imprinted card

Reply:
hmm, now I've got an out of bounds exception with Card.IsRemembered+IsNotImprinted